packages/cacti-plugin-consortium-static/src/main/typescript/repository/new-consortium-repository.ts

```diff
@@ -0,0 +1,199 @@
+import {
+  Logger,
+  Checks,
+  LogLevelDesc,
+  LoggerProvider,
+} from "@hyperledger/cactus-common";
+
+import {
+  CactusNode,
+  ConsortiumDatabase,
+  ICactusPluginOptions,
+  Ledger,
+  PluginInstance,
+} from "@hyperledger/cactus-core-api";
+import { PolicyGroup } from "../policy-model/policy-group";
+import MerkleTree from "merkletreejs";
+import { JWK } from "jose";
+import { verifyOrganization } from "../utils";
+
+export interface INewConsortiumRepositoryOptions {
+  logLevel?: LogLevelDesc;
+  db: ConsortiumDatabase;
+  entitiesJWK: { [key: string]: JWK };
+  rootPolicyGroup?: PolicyGroup;
+  packageConfigs?: { [key: string]: unknown };
+
+  //data about self
+  node: CactusNode;
+  ledgers: Ledger[];
+  pluginInstances: PluginInstance[];
+  memberId: string;
+}
+
+/**
+ * Class responsible for making it convenient for developers to query the
+ * `ConsortiumDatabase` model type which is a flat data structure storing
+ * all the different types of entities for Consortium representation such as
+ * `CactusNode`, `ConsortiumMember`, `Ledger` etc..
+ */
+export class NewConsortiumRepository {
+  public static readonly CLASS_NAME = "NewConsortiumRepository";
+
```

Comment:
@eduv09 Please rename this as well, I'm guessing it's a leftover from before we came up with the current name.